components/molecules/Table.jsx

```diff
@@ -164,18 +164,20 @@ class Table extends Component {
     }
 
     render() {
-        const { data, className, children, size, isStriped, isInverse } = this.props;
-        return (
-            <table
-                className={cx('gds-table', className, {
-                    [`gds-table--${size}`]: size,
-                    [`gds-table--inverse`]: isInverse,
-                    [`gds-table--striped`]: isStriped && !isInverse,
-                    [`gds-table--inverse-striped`]: isStriped && isInverse
-                })}>
-                {data ? this.renderTable(this.props) : children}
-            </table>
-        );
+        const { data, className, children, size, isStriped, isInverse, isResponsive } = this.props;
+        const WithTable = () => (<table
```

Comment:
I don't think it is necessary to separate the component. This could be simplified by always displaying the wrapping `<div>` and just add the responsive className conditionally, something like:

```jsx
<div className={ isResponsive ? 'gds-table--responsive' : '' } >
    <table
        className={cx('gds-table', className, {
            [`gds-table--${size}`]: size,
            ['gds-table--inverse']: isInverse,
            ['gds-table--striped']: isStriped && !isInverse,
            ['gds-table--inverse-striped']: isStriped && isInverse
        })}>
        {data ? this.renderTable(this.props) : children}
    </table>
</div>
```

Unless anyone thinks there would be side effects from doing this. Any thoughts?

Reply:
It's nice to not have an extra div when it's not needed, but I don't think it will cause any side effects.

Normally, I'd prefer doing something like the following in this type of situation:

```jsx
renderTable() {
    const { data, className, children, size, isStriped, isInverse } = this.props;
    return (
        <table
            className={cx('gds-table', className, {
                [`gds-table--${size}`]: size,
                [`gds-table--inverse`]: isInverse,
                [`gds-table--striped`]: isStriped && !isInverse,
                [`gds-table--inverse-striped`]: isStriped && isInverse
            })}>
            {data ? this.renderTableContent(this.props) : children} // NOTE: I renamed this from renderTable to renderTableContent
        </table>
    );
}

render() {
    const { isResponsive } = this.props;
    return isResponsive ? (
        <div className="gds-table--responsive">{this.renderTable()}</div>
    ) : (
        this.renderTable()
    );
}
```